File: src/pratice/binarysearch/수찾기1920.java
package pratice.binarysearch;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;

public class 수찾기1920 {
    public static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    public static int N, M;
    public static int[] A, B;
    public static void main(String[] args) throws IOException {
        input();
        Arrays.sort(A);
        for (int i = 0; i < M; i++) {
            pro(B[i]);
        }
    }

    private static void pro(int num) {
        int L = 0 ;
        int R = A.length - 1;
        boolean check = false;
        while(L <= R){
            int mid = (L + R) / 2;
            if(A[mid] == num){
                System.out.println(1);
                return;
            }else if(A[mid] < num){
                L = mid + 1;
            }else{
                R = mid - 1;
            }
        }
        System.out.println(0);
    }

    private static void input() throws IOException {
        N = Integer.parseInt(br.readLine());
        A = new int[N];
        String[] spilt = br.readLine().split(" ");
        for (int i = 0; i < N; i++) {
            A[i] = Integer.parseInt(spilt[i]);
        }

        M = Integer.parseInt(br.readLine());
        B = new int[M];
        spilt = br.readLine().split(" ");
        for (int i = 0; i < M; i++) {
            B[i] = Integer.parseInt(spilt[i]);
        }
    }
}

File: src/pratice/binarysearch/수찾기1920.kt
package pratice.binarysearch

fun main() {
    val N = readln().toInt()
    val a = readln().split(" ").map { it.toInt() }.toMutableList()
    a.sort()

    var M = readln().toInt()
    val b = readln().split(" ").map { it.toInt() }.toMutableList()

    for (num in b) {
        println(binarySearch(a, num))
    }
}

private fun binarySearch(arr: MutableList<Int>, num: Int): Int {
    var L = 0
    var R = arr.size - 1
    while (L <= R) {
        val mid = (L + R) / 2
        if (arr[mid] == num) {
            return 1
        } else if (arr[mid] < num) {
            L = mid + 1
        } else {
            R = mid - 1
        }
    }
    return 0
}
